app/experimenter/experiments/api/v5/queries.py

```diff
@@ -53,7 +54,26 @@ def resolve_outcomes(root, info):
         return Outcomes.all()
 
     def resolve_targeting_config_slug(root, info):
-        return root._text_choices_to_label_value_list(NimbusExperiment.TargetingConfig)
+        resolved_configs = []
+        application_value_by_slug = dict(
+            (member.value, value)
+            for (value, member) in NimbusExperiment.Application.__members__.items()
+        )
+        for (value, member) in NimbusExperiment.TargetingConfig.__members__.items():
+            targeting_config = NimbusExperiment.TARGETING_CONFIGS.get(member.value, None)
+            application_value = (
+                targeting_config.application
+                and application_value_by_slug[targeting_config.application.slug]
+                or ""
+            )
+            resolved_configs.append(
+                NimbusExperimentTargetingConfigSlugChoice(
+                    label=targeting_config.name,
+                    value=value,
+                    application_value=application_value,
+                )
+            )
+        return resolved_configs
```

Comment:
Trying to think of a more elegant way to do this. But, we've got a bit of a tangle here between our dataclasses and `models.TextChoices` members.

We use slugs in the database. But, forms & mutations use `name` from `TextChoices` members. These are not the same values. (i.e. targeting slug "first_run_win1903" is represented by a choice named `TARGETING_FIRST_RUN_WINDOWS_1903_NEWER` and application slug "firefox-desktop" corresponds to choice `DESKTOP`)

Reply:
Hold on let's rethink this for a second.  In the UI, we only care about application or targeting `name`, we don't use `slug` at all for either right?

So if in our targeting configs we say

```py
    application = Application.DESKTOP.name
```

and just pass that off verbatim to the frontend, we do the filtering on the frontend, frontends `name` back, we already resolve that back to the value, then we don't need to do this transformation right?

Reply:
I can do that, though I figured that just pushes the hack around a bit. At some point if we need to refer to the rest of the application associated in targeting, we need to do a different mapping. But, for the sake of YAGNI in this current PR, just using the name would be cleaner.

Reply:
Yeah exactly we can deal with that later if we need to.

Reply:
Oh, yeah, and I was also hoping that specifying `application: ApplicationConfig` would get some type-checking benefit. But that doesn't seem to do what I thought it would

Reply:
Reworked this to use a list of choice names specified for each targeting config